### PR TITLE
[FIXED JENKINS-26596]: Update view when restarting build from view

### PR DESCRIPTION
**Pull request by jeckste**
[JENKINS-26596](https://issues.jenkins-ci.org/browse/JENKINS-26596)

### CODE REVIEW

**Comment by Idanatcox** [comment]
Validated this fix on my environment +  #68 scenario @patbos describes passed successfully.
thanks

**Comment by stuartstevenson** [comment]
hey folks, any update on the status of this? Is there anything holding it up from being merged?

**Comment by Idanatcox** [comment]
hey @stuartstevenson , we've been using this fix for months now flawlessly.
I thought the maintainer - centrum is responsible for merging it in... but they are not responding for months now

**Comment by stuartstevenson** [comment]
@Idanatcox thanks for the reply. I think I will email the mailing lists to get access to merge this in if I don't get a response soon

**Comment by recena** [comment]
@jeckste Could you accept the [JIRA issue](https://issues.jenkins-ci.org/browse/JENKINS-26596)?

**Comment by jeckste** [comment]
@recena I assigned the issue to myself

**Comment by recena** [comment]
@jeckste Could you assign yourself this issue [JENKINS-19870](https://issues.jenkins-ci.org/browse/JENKINS-19870)?

**Comment by jeckste** [comment]
@recena I have assigned this issue to myself as well

**Comment by recena** [comment]
@dalvizu and I will review this PR.

**Comment by Idanatcox** [comment]
Consider the following scenario:
A -> B -> C

trigger A,B,C
rerun B
current result:
B updated with the latest build info
C(or any downstream) status will be reset to NOT_BUILT
alternative result:
B updated with the latest build info
C(or any downstream) status should not reset

I just wanted to point it out since we've been using the pipeline for a few months and came across this scenario.
Do you mind considering what should be the right behavior?

File: src/main/java/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineView.java [hunk 1]
```diff
@@ -428,8 +429,8 @@ public int rerunBuild(final String externalizableId) {
         LOGGER.fine("Running build again: " + externalizableId); //$NON-NLS-1$
         final AbstractBuild<?, ?> triggerBuild = (AbstractBuild<?, ?>) Run.fromExternalizableId(externalizableId);
         final AbstractProject<?, ?> triggerProject = triggerBuild.getProject();
-        final Future<?> future = triggerProject.scheduleBuild2(triggerProject.getQuietPeriod(), new MyUserIdCause(),
-                removeUserIdCauseActions(triggerBuild.getActions()));
+        final Future<?> future = triggerProject.scheduleBuild2(triggerProject.getQuietPeriod(), null,
```
Comment (dalvizu) [inline src/main/java/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineView.java]: Why is a new MyUserIdCause() not added? I believe this is necessary to set the user id to the current user who clicked the 'retrigger' button. If it is not set, the user will come from the last run, which isn't necessarily true.

**Comment by dalvizu** [comment]
I believe this is fixed in #89 - can you re-test on latest master?

**Comment by Idanatcox** [comment]
I confirmed that #89 fixes this issue too.
There is no need to merge both #73 and #89 
Can you please close this PR?

**Comment by dalvizu** [comment]
Excellent! Thank you!

**Pull request closed without merging**